### PR TITLE
Update ver, named vols, use latest

### PR DESCRIPTION
**Pull request by BNolet**
* Version of docker-compose updated to 3.0
* Changed volumes to named ones
* Now uses latest version of zulip container always

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -19,13 +19,13 @@ services:
...
-      - '/opt/docker/zulip/rabbitmq:/var/lib/rabbitmq:rw'
+      - 'rabbitmq-vol:/var/lib/rabbitmq:rw'
   redis:
     image: 'quay.io/sameersbn/redis:latest'
     volumes:
-      - '/opt/docker/zulip/redis:/var/lib/redis:rw'
+      - 'redis-vol:/var/lib/redis:rw'
   zulip:
-    image: 'zulip/docker-zulip:1.9.0'
+    image: 'zulip/docker-zulip'
```
Comment (galexrt) [inline docker-compose.yml]: This is bad for production, you don't just want your Zulip instance to update when you don't want it to.

Comment (BNolet) [reply docker-compose.yml]: Alright, will put it back

**Comment by BNolet** [comment]
Should be good now

**Comment by timabbott** [comment]
I rebased this to do some cleanup of the history around version changes and rebase onto master (where the one version change that wasn't reverted has already been fixed).  

The one concern I have is about how the named volume system works for folks upgrading; will `redis-vol` still have a default data dir path of `/opt/docker/zulip/redis`?  If not, we should supply that.

**Comment by timabbott** [comment]
@BNolet @andersk and I were just looking at this earlier today.  I think we do want to merge this, but we can't do so without some work on documentation/announcing the change, because it'll potentially break access to files when users upgrade the project.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-version: '2'
+version: '3.0'
```
Comment (timabbott) [inline docker-compose.yml]: I don't believe this change is required for the named volumes feature; can you explain why you made it?  It breaks using this with the default version of Docker in Ubuntu Xenial

Comment (BNolet) [reply docker-compose.yml]: No real reason besides moving forward, I can revert this if you like.

**Comment by BNolet** [comment]
If you like, I can work on the documentation if you need, just point me in the right direction of what you want.

**Comment by timabbott** [comment]
For documentation: I think we mostly just need to make a section on "upgrading", link to it prominently from the top of the README.md next to the versions, and have it include both a note on this as well the existing note on upgrading from the older 1.8.1 project.

File: docker-compose.yml (hunk 2 above)
Comment (timabbott) [reply docker-compose.yml]: Yeah, let's revert this; I think the purpose of this version tag is to indicate what features we're requiring, and this doesn't have us use any features that are new in 3.0, so we should keep it at 2.0 for compatibility with as many installations as will work with it.

Comment (BNolet) [reply docker-compose.yml]: Reverted to 2

**Comment by maltokyo** [comment]
Most people will anyway edit their compose file, does anyone use it "out of the box" for critical applications (not just zulip)? People will want to add their own network (reverse proxy etc) and check that the volumes are stored in line with the other dockers running on their machine already. After I remove all the secrets etc, I will post mind here, as an example of what I would expect many people doing.

Wouldn't the most helpful and least destructive way to upgrade the compose file be to not include a `docker-compose.yml` file at all in the clone, and rather include `docker-compose.yml.sample` with all of the content in this PR?  Overwriting a `docker-compose.yml` unwittingly will always cause headaches.  Better to provide it as a sample and have the admin duplicate that and knowingly rename it to `docker-compose.yml`or just use it as a reference to update their existing one manually.  Their old one will still work without this "upgraded", so why force it?

**Pull request closed without merging**